### PR TITLE
issue-427: Removed #ifndef against #pragma once in all .h files in Simulator

### LINKED ISSUE #427: Remove either '#ifdef' directives or '#pragma once'
A few files like `Global.h` have both `#pragma once` and `#ifdef` directives defined in the same file. Both these directives are intended to prevent multiple inclusions of header files. As per C++ standard `#pragma once` is usually sufficient to prevent multiple inclusions by itself. However, we must aim to clean these and keep which serves best based on the use case.

### PR DESCRIPTION
**Pull request by Sanchit-kumar**
Closes #427 
Following changes made in .h files present in Simulator:
1. Added #pragma once if missing
2. Removed #ifndef if required
3. Removed extra tab for the indentation if required

### CODE REVIEW

File: Simulator/Utils/Book.h [hunk 1]
```diff
@@ -20,7 +20,7 @@
  */
 
 
-#ifndef __BOOK_H__
+#pragma once
 #define __BOOK_H__
```
Comment (stiber) [inline Simulator/Utils/Book.h:24]: Should be deleted.

File: Simulator/Utils/Coordinate.h [hunk 2]
```diff
@@ -6,7 +6,7 @@
  * @brief A container for 2-dimensional coordinates.
  */
 
-#ifndef _COORDINATE_H_
+#pragma once
 #define _COORDINATE_H_
```
Comment (stiber) [inline Simulator/Utils/Coordinate.h:10]: Delete.

File: Simulator/Utils/Global.h [hunk 3]
```diff
@@ -9,48 +9,47 @@
...
-   // #define DEBUG_LOG(__lvl, __x) { if(__lvl & d_debug_mask[0]) { __x } }
-   #else
-      #define DEBUG_LOG(__lvl, __x)                                                                \
-         {                                                                                         \
-            if (__lvl & g_debug_mask) {                                                            \
-               __x                                                                                 \
-            }                                                                                      \
-         }
-   #endif
+#define _GLOBAL_H_
```
Comment (stiber) [inline Simulator/Utils/Global.h:12]: Delete.

File: Simulator/Utils/Inputs/FSInput.h [hunk 4]
```diff
@@ -18,10 +18,9 @@
 
 #pragma once
 
-#ifndef _FSINPUT_H_
-   #define _FSINPUT_H_
+#define _FSINPUT_H_
```
Comment (stiber) [inline Simulator/Utils/Inputs/FSInput.h:21]: Delete.

File: Simulator/Utils/Inputs/GpuSInputPoisson.h [hunk 5]
```diff
@@ -13,12 +13,11 @@
 
 #pragma once
 
-#ifndef _GPUSINPUTPOISSON_H_
-   #define _GPUSINPUTPOISSON_H_
+#define _GPUSINPUTPOISSON_H_
```
Comment (stiber) [inline Simulator/Utils/Inputs/GpuSInputPoisson.h:16]: Delete.

File: Simulator/Utils/Inputs/GpuSInputRegular.h [hunk 6]
```diff
@@ -12,10 +12,9 @@
 
 #pragma once
 
-#ifndef _GPUSINPUTREGULAR_H_
-   #define _GPUSINPUTREGULAR_H_
+#define _GPUSINPUTREGULAR_H_
```
Comment (stiber) [inline Simulator/Utils/Inputs/GpuSInputRegular.h:15]: Delete.

File: Simulator/Utils/Inputs/HostSInputPoisson.h [hunk 7]
```diff
@@ -13,10 +13,9 @@
 
 #pragma once
 
-#ifndef _HOSTSINPUTPOISSON_H_
-   #define _HOSTSINPUTPOISSON_H_
+#define _HOSTSINPUTPOISSON_H_
```
Comment (stiber) [inline Simulator/Utils/Inputs/HostSInputPoisson.h:16]: Delete.

File: Simulator/Utils/Inputs/HostSInputRegular.h [hunk 8]
```diff
@@ -12,10 +12,9 @@
 
 #pragma once
 
-#ifndef _HOSTSINPUTREGULAR_H_
-   #define _HOSTSINPUTREGULAR_H_
+#define _HOSTSINPUTREGULAR_H_
```
Comment (stiber) [inline Simulator/Utils/Inputs/HostSInputRegular.h:15]: Delete.

File: Simulator/Utils/Inputs/ISInput.h [hunk 9]
```diff
@@ -10,13 +10,12 @@
 
 #pragma once
 
-#ifndef _ISINPUT_H_
-   #define _ISINPUT_H_
+#define _ISINPUT_H_
```
Comment (stiber) [inline Simulator/Utils/Inputs/ISInput.h:13]: Delete.

File: Simulator/Utils/Inputs/SInputPoisson.h [hunk 10]
```diff
@@ -17,11 +17,10 @@
 
 #include <vector>
 
-#ifndef _SINPUTPOISSON_H_
-   #define _SINPUTPOISSON_H_
+#define _SINPUTPOISSON_H_
```
Comment (stiber) [inline Simulator/Utils/Inputs/SInputPoisson.h:20]: Delete.

File: Simulator/Utils/Inputs/SInputRegular.h [hunk 11]
```diff
@@ -16,10 +16,9 @@
 
 #include <vector>
 
-#ifndef _SINPUTREGULAR_H_
-   #define _SINPUTREGULAR_H_
+#define _SINPUTREGULAR_H_
```
Comment (stiber) [inline Simulator/Utils/Inputs/SInputRegular.h:19]: Delete.

File: Simulator/Utils/Matrix/Matrix.h [hunk 12]
```diff
@@ -15,7 +15,7 @@
  * using only Matrix objects.
 */
 
-#ifndef _MATRIX_H_
+#pragma once
 #define _MATRIX_H_
```
Comment (stiber) [inline Simulator/Utils/Matrix/Matrix.h:19]: Delete.

File: Simulator/Utils/Matrix/MatrixFactory.h [hunk 13]
```diff
@@ -6,7 +6,7 @@
  * @brief Deserializes Matrices from XML
  */
 
-#ifndef _MATRIXFACTORY_H_
+#pragma once
 #define _MATRIXFACTORY_H_
```
Comment (stiber) [inline Simulator/Utils/Matrix/MatrixFactory.h:10]: Delete.

File: Simulator/Utils/Matrix/SparseMatrix.h [hunk 14]
```diff
@@ -6,7 +6,7 @@
  * @brief An efficient implementation of a dynamically-allocated 2D sparse array.
  */
 
-#ifndef _SPARSEMATRIX_H_
+#pragma once
 #define _SPARSEMATRIX_H_
```
Comment (stiber) [inline Simulator/Utils/Matrix/SparseMatrix.h:10]: Delete.

File: Simulator/Utils/Matrix/VectorMatrix.h [hunk 15]
```diff
@@ -7,7 +7,7 @@
  */
 
 
-#ifndef _VECTORMATRIX_H_
+#pragma once
 #define _VECTORMATRIX_H_
```
Comment (stiber) [inline Simulator/Utils/Matrix/VectorMatrix.h:11]: Delete.

File: Simulator/Utils/ParseParamError.h [hunk 16]
```diff
@@ -7,7 +7,7 @@
  * 
  */
 
-#ifndef _PARSEPARAMERROR_H_
+#pragma once
 #define _PARSEPARAMERROR_H_
```
Comment (stiber) [inline Simulator/Utils/ParseParamError.h:11]: Delete.

File: Simulator/Utils/RNG/MersenneTwister_d.h [hunk 17]
```diff
@@ -17,12 +17,12 @@
 
 // #ifndef MERSENNETWISTER_H
 //#define MERSENNETWISTER_H
-#ifndef mersennetwister_h
-   #define mersennetwister_h
+#pragma once
+#define mersennetwister_h
```
Comment (stiber) [inline Simulator/Utils/RNG/MersenneTwister_d.h:21]: Delete.

File: Simulator/Utils/RNG/Norm.h [hunk 18]
```diff
@@ -18,7 +18,7 @@
 //
 //
 
-#ifndef _NORM_H_
+#pragma once
 #define _NORM_H_
```
Comment (stiber) [inline Simulator/Utils/RNG/Norm.h:22]: Delete.

File: Simulator/Utils/Timer.h [hunk 19]
```diff
@@ -6,7 +6,7 @@
  * @brief Timer class, from CSS432
  */
 
-#ifndef _TIMER_H_
+#pragma once
 #define _TIMER_H_
```
Comment (stiber) [inline Simulator/Utils/Timer.h:10]: Delete.

File: Simulator/Utils/Util.h [hunk 20]
```diff
@@ -8,13 +8,10 @@
 
 #pragma once
 
-#ifndef _UTIL_H_
-   #define _UTIL_H_
+#define _UTIL_H_
```
Comment (stiber) [inline Simulator/Utils/Util.h:11]: Delete.

**Review by stiber — CHANGES_REQUESTED** [review]
Some "orphan" `#define`s left over that are no longer needed.

**Comment by stiber** [comment]
I think one more file waiting (see unresolved conversation).

**Comment by stiber** [comment]
It seems that this PR is failing our style check. If it isn't obvious what the issues are, you might try running a recent version of `clang-format` to generate a reformatted version of the file(s) in question, and then do a `diff` to see what it's unhappy about. Sometimes, this is extra spaces or tabs instead of spaces or the like. Our GitHub Pages has some more documentation about this.

**Comment by Sanchit-kumar** [comment]
I missed a few files for code style check. Updated the code style with respect to .clang-format file.

**Review by stiber — APPROVED** [review]

**Pull request merged**